File: go/docmost/models/update_workspace_dto.go
package models

// UpdateWorkspaceDto from OpenAPI.
type UpdateWorkspaceDto struct {
	Name *string `json:"name,omitempty"`
	Hostname *string `json:"hostname,omitempty"`
	Description *string `json:"description,omitempty"`
	Logo *string `json:"logo,omitempty"`
	EmailDomains []string `json:"emailDomains,omitempty"`
	EnforceSso *bool `json:"enforceSso,omitempty"`
	EnforceMfa *bool `json:"enforceMfa,omitempty"`
	RestrictApiToAdmins *bool `json:"restrictApiToAdmins,omitempty"`
	AiSearch *bool `json:"aiSearch,omitempty"`
	GenerativeAi *bool `json:"generativeAi,omitempty"`
	DisablePublicSharing *bool `json:"disablePublicSharing,omitempty"`
}

File: c#/src/Docmost.Sdk/Models/UpdateWorkspaceDto.cs
namespace Docmost.Sdk.Models;

public class UpdateWorkspaceDto
{
    [System.Text.Json.Serialization.JsonPropertyName("name")]
    public string? Name { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("hostname")]
    public string? Hostname { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("description")]
    public string? Description { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("logo")]
    public string? Logo { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("emailDomains")]
    public List<string>? EmailDomains { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("enforceSso")]
    public bool? EnforceSso { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("enforceMfa")]
    public bool? EnforceMfa { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("restrictApiToAdmins")]
    public bool? RestrictApiToAdmins { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("aiSearch")]
    public bool? AiSearch { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("generativeAi")]
    public bool? GenerativeAi { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("disablePublicSharing")]
    public bool? DisablePublicSharing { get; set; }

}
